osconfig_tests/test_suites/package_management/package_management.go

```diff
@@ -352,7 +361,7 @@ func runPackageInstallTest(ctx context.Context, testCase *junitxml.TestCase, tes
 	testCase.Logf("Agent installed successfully")
 
 	// read the serial console once
-	if err = testSetup.vf(inst, testSetup.vstring, 1, 10*time.Second, 60*time.Second); err != nil {
+	if err = testSetup.vf(inst, testSetup.vstring, 1, 10*time.Second, 1200*time.Second); err != nil {
```

Comment:
This is a long time to wait and I'm concerned that it may mean that failing tests take 20 times longer to actually fail. Can this be scoped just to Windows, if that's where the extra time is needed? Can we use bigger test instances to make this faster? Are there failure signals that can be checked to ensure that the test is waiting on a success and not waiting to time out 20 minutes after a failure was reported?

Reply:
> This is a long time to wait and I'm concerned that it may mean that failing tests take 20 times longer to actually fail. Can we use bigger test instances to make this faster?
windows startup is terribly long. Digging more into it, I found that the time is majorly spent at 2 different stages. 1) windows boot and startup script installation; 2) fetching windows updates.
I also looked at the resource utilization of the test instances and found that with the smaller instances (1 cpu/4gb ram - standard/1), the CPU utilization is almost always 100%. Moving to bigger instances should definitely help in improving the boot time(I have just kicked off a test with bigger instance(4 cpu/8gigs - standard/4); would post the results shortly)
Surprisingly, it also helped in faster fetching of windows updates. [logs](https://pantheon.corp.google.com/logs/viewer?resource=gce_instance%2Finstance_id%2F6364734612471780596&project=subratp-playground&minLogLevel=0&expandAll=false&timestamp=2019-04-10T17:53:57.345000000Z&customFacets=&limitCustomFacetWidth=true&interval=PT1H&scrollTimestamp=2019-04-10T17:50:43.159815000Z&dateRangeStart=2019-04-10T16:49:28.312Z&dateRangeUnbound=forwardInTime)

I also tested with even bigger instance(8 cpu), i dont see an improvement over standard/4. 
I will use standard/4 as test instances.

> Can this be scoped just to Windows, if that's where the extra time is needed?
this can done. I can add a new member to packagemanagementestsetup, so while adding a testsetup, a custom timeout for verification string can be provided for individual tests.

> Are there failure signals that can be checked to ensure that the test is waiting on a success and not waiting to time out 20 minutes after a failure was reported?
Currently, there is no strong failure signal that we get from the tests and i did not want it on purpose for the reason that I wanted to assert on success signals instead of failure signal. while performing all the manual tests, I realized that it was hard to define strong failure signals because there were multiple reasons for a test to fail(these were reasons that i could figure out, there might be others as well). So, instead i chose to have a strong success signal.